### PR TITLE
Update PatternFly to latest versions

### PR DESCRIPTION
**Pull request by dtaylor113**
package.json
```
   "dependencies": {
     ...
-    "@patternfly/patternfly": "2.65.3",
-    "@patternfly/react-catalog-view-extension": "1.4.11",
-    "@patternfly/react-charts": "5.2.2",
-    "@patternfly/react-core": "3.140.11",
-    "@patternfly/react-table": "2.24.41",
-    "@patternfly/react-tokens": "2.7.10",
-    "@patternfly/react-topology": "2.11.27",
-    "@patternfly/react-virtualized-extension": "1.3.40",
+    "@patternfly/patternfly": "2.71.1",
+    "@patternfly/react-charts": "5.3.17",
+    "@patternfly/react-core": "3.153.9",
+    "@patternfly/react-table": "2.28.28",
+    "@patternfly/react-tokens": "2.8.11",
+    "@patternfly/react-topology": "2.14.47",
+    "@patternfly/react-virtualized-extension": "1.4.48",
+    "@patternfly/react-catalog-view-extension": "1.4.47",
     ...
```

This PR also includes fixes for:
- Notification Drawer: Notification Drawer is labeled `beta` in PF meaning breaking changes can occur. This PR updates OpenShift's usage of NotificationDrawer to adhere to latest PF implementation.
- CatalogTile/Item: #4168 set `maxDescriptionLength` to -1 thus nullifying it's usage.   In latest PF packages/versions `maxDescriptionLength` has been removed entirely.   If was an optional prop, so I'm not sure if removing it from PF was correct procedure, as removing it entirely could be considered a breaking change.  This was discussed [here](https://github.com/patternfly/patternfly-react/pull/3830#pullrequestreview-366463600) and the change in PF was made.
This PR removes the property from OpenShift

### CODE REVIEW

**Comment by spadgett** [comment]
/hold
@dtaylor113 you shouldn't remove the yarn lockfile

**Comment by spadgett** [comment]
/hold cancel
sorry my mistake

**Comment by spadgett** [comment]
/approve
/assign @jcaianirh 

@jcaianirh please verify the drawer changes, thanks!

**Comment by jcaianirh** [comment]
Is it possible to get the fix for this in too ? 

https://github.com/patternfly/patternfly-react/pull/4060#issuecomment-614645277

@patternfly/react-core@3.153.9

**Comment by dtaylor113** [comment]
> Is it possible to get the fix for this in too ?
> 
> [patternfly/patternfly-react#4060 (comment)](https://github.com/patternfly/patternfly-react/pull/4060#issuecomment-614645277)
> 
> @patternfly/react-core@3.153.9

Hi @jcaianirh, updated patternfly/react-core to 3.153.9

**Comment by rhamilto** [comment]
/hold

This breaks terminal full screen.  @jcaianirh fixed this once.

Screen Shot 2020-04-16 at 1 48 06 PM

**Comment by rhamilto** [comment]
Further, the layout is busted at viewports >= 1800px.
Screen Shot 2020-04-16 at 2 28 06 PM

Definitely could use @jcaianirh's expertise here since he's familiar with the component.

**Comment by rhamilto** [comment]
Suspiciously eyeing https://github.com/openshift/console/pull/4414/files

**Comment by jcaianirh** [comment]
@rhamilto @dtaylor113 the css fix for the terminal is intact. It looks like we have to override a new css variable in _overrides.scss: 

`  
  .pf-c-drawer__content {
    --pf-c-drawer__content--ZIndex: auto;
  }
`
Tested this with full page terminal and looks good.

**Comment by dtaylor113** [comment]
> override a new css variable in _overrides.scss:
> 
> ` .pf-c-drawer__content { --pf-c-drawer__content--ZIndex: auto; }`
> Tested this with full page terminal and looks good.

@jcaianirh , @rhamilto added override, terminal now goes to fullscreen and can click to 'Collapse'
image

**Comment by jcaianirh** [comment]
@dtaylor113 @rhamilto one more thing, i see double scroll bars with long alert list. 
Fix is in .notification-drawer.scss:
`.co-notification-drawer {
  z-index: 10;
  **height: fit-content;**
}`
Add the height css here

**Comment by dtaylor113** [comment]
Also, what about the busted layout at viewports >= 1800px that @rhamilto reported?

**Comment by dtaylor113** [comment]
Added fix in .notification-drawer.scss but wasn't able to verify in cluster (not enough alerts in notification drawer to cause scroll).  Got errors trying to create a new cluster, need to download pull secret.  I'll try in morning.

**Comment by jcaianirh** [comment]
@dtaylor113 , i've been retesting, and I think this is a better fix:
`  .co-notification-drawer { z-index: 10; overflow-y: hidden !important;}
`

**Comment by dtaylor113** [comment]
> .co-notification-drawer { z-index: 10; overflow-y: hidden !important;}

Hi @jcaianirh, ok updated to ^^

**Comment by rhamilto** [comment]
@dtaylor113 please squash.

**Comment by dtaylor113** [comment]
@rhamilto, squashed :-)

**Comment by rhamilto** [comment]
Screen Shot 2020-04-17 at 9 45 41 AM

Cosmetic issue at viewport sizes 1800px+ with the drawer heading.  Trying to identify a fix.

**Comment by rhamilto** [comment]
Adding the following to `_notification-drawer.scss` appears to fix the bug:
```css
.pf-c-notification-drawer__header {
  // fix upstream bug where header overlays divider
  @media(min-width: 1800px) {
    margin-left: 1px;
  }
}
```
Screen Shot 2020-04-17 at 10 02 17 AM

**Comment by dtaylor113** [comment]
Thanks @rhamilto, added `.pf-c-notification-drawer__header`

**Comment by rhamilto** [comment]
Hmm.  The hard-coded media query gives me pause.  The correct fix is probably this:

```css
.pf-c-drawer.pf-m-inline .pf-c-notification-drawer__header {
  // fix upstream bug where header overlays divider
  margin-left: 1px;
}
```

**Comment by dtaylor113** [comment]
Hi @rhamilto, updated to `.pf-c-drawer.pf-m-inline .pf-c-notification-drawer__header` -thanks

**Comment by rhamilto** [comment]
/lgtm

@jcaianirh, mind giving this another look?  If you're good with it, remove the hold.

File: frontend/public/style/_overrides.scss [hunk 1]
```diff
@@ -466,3 +466,5 @@ button.pf-c-dropdown__menu-item.pf-m-disabled {
   // enables tooltips for disabled menu items
   pointer-events: auto;
 }
+
+.pf-c-drawer__content { --pf-c-drawer__content--ZIndex: auto; }
```
Comment (rhamilto) [inline frontend/public/style/_overrides.scss:470]: Sorry.  On second thought, I think this should go in `_notification-drawer.scss` so all the drawer scss is in the same file.

Comment (dtaylor113) [reply frontend/public/style/_overrides.scss]: @rhamilto, moved to `_notification-drawer.scss` and used multiple lines :-)

**Comment by jcaianirh** [comment]
Added these css mods to pf issues to get them fixed upstream: https://github.com/patternfly/patternfly/issues/2977

**Comment by spadgett** [comment]
/hold
for #5100

**Pull request merged**